pkg/types/types.go

```diff
@@ -20,3 +24,27 @@ func (v JobID) String() string {
 func (v RunID) String() string {
 	return strconv.FormatUint(uint64(v), 10)
 }
+
+type key string
+
+const (
+	KeyJobID = key("job_id")
+	KeyRunID = key("run_id")
+)
+
+// JobIDFromContext is a helper to get the JobID, this is useful
+// for plugins which need to know which job they are running.
+// Not all context object everywhere have this set, but this is
+// guaranteed to work in TargetManagers, TestSteps and Reporters
+func JobIDFromContext(ctx xcontext.Context) (JobID, bool) {
+	v, ok := ctx.Value(KeyJobID).(JobID)
+	return v, ok
+}
```

Comment:
I understand this sounds very controversial, and personally I do not really have an opinion. But...

I somewhat do not like the idea of checking `ok` **each time** we extract `JobID`. So, may be panicking is fine in this case?

Do not take this comment seriously, it is just a crazy thought.

Reply:
Yeah I also though about this. I ended up with this:
```
// guaranteed to work here
jobID, _ := types.JobIDFromContext(ctx)
```